File: backend-csharp/Services/OrderService.cs
using DeliveryHub.Models;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.Configuration;

namespace DeliveryHub.Services;

public class OrderService
{
    private readonly DeliveryHubContext _context;
    private readonly string _connectionString;

    public OrderService(DeliveryHubContext context, IConfiguration configuration)
    {
        _context = context;
        _connectionString = configuration.GetConnectionString("DefaultConnection") ?? "";
    }

    public Order CreateOrder(Order order)
    {
        order.Status = "PENDING";
        order.CreatedAt = DateTime.Now;
        order.UpdatedAt = DateTime.Now;
        _context.Orders.Add(order);
        _context.SaveChanges();
        return order;
    }

    public Order? GetOrderById(int id)
    {
        try
        {
            var order = _context.Orders.Find(id);
            if (order == null)
            {
                throw new Exception("주문을 찾을 수 없습니다: " + id);
            }
            return order;
        }
        catch (InvalidOperationException e)
        {
            Console.WriteLine("잘못된 작업: " + e.Message);
            return null;
        }
        catch (Exception e)
        {
            Console.WriteLine("에러 발생: " + e.Message);
            return null;
        }
    }

    public List<Order> GetAllOrders()
    {
        return _context.Orders.ToList();
    }

    public Order? UpdateOrderStatus(int id, string status)
    {
        var order = _context.Orders.Find(id);
        if (order != null)
        {
            order.Status = status;
            order.UpdatedAt = DateTime.Now;
            _context.SaveChanges();
        }
        return order;
    }

    public List<Order> SearchOrders(string keyword)
    {
        return _context.Orders
            .Where(o => o.CustomerName.Contains(keyword) || o.StoreName.Contains(keyword))
            .ToList();
    }

    public async Task ProcessOrderAsync(int orderId)
    {
        await Task.Run(async () =>
        {
            var order = GetOrderById(orderId);
            await Task.Delay(3000);
            UpdateOrderStatus(orderId, "PROCESSING");
        });
    }

    public int CalculateTotal(List<Order> orders)
    {
        int total = 0;
        foreach (var order in orders)
        {
            total += order.TotalPrice;
        }
        return total;
    }
}

File: backend-python/app/services/order_service.py
from app.models.order import Order
from datetime import datetime
import sqlite3
import threading

# 메모리 DB
orders_db = []
next_id = 1


def create_order(order: Order) -> Order:
    global next_id
    order.id = next_id
    next_id += 1
    order.status = "PENDING"
    order.created_at = datetime.now()
    order.updated_at = datetime.now()
    orders_db.append(order)
    return order


def get_order_by_id(order_id: int) -> Order:
    for order in orders_db:
        if order.id == order_id:
            return order
    return None


def get_all_orders() -> list:
    return orders_db


def update_order_status(order_id: int, status: str) -> Order:
    for order in orders_db:
        if order.id == order_id:
            order.status = status
            order.updated_at = datetime.now()
            return order
    return None


def search_orders(keyword: str) -> list:
    results = []
    for order in orders_db:
        if keyword in order.customer_name or keyword in order.store_name:
            results.append(order)
    return results


def delete_order(order_id: int) -> bool:
    global orders_db
    for i in range(len(orders_db)):
        if orders_db[i].id == order_id:
            del orders_db[i]
            return True
    return False


def process_order_async(order_id: int):
    def worker():
        try:
            order = get_order_by_id(order_id)
            import time
            time.sleep(3)
            update_order_status(order_id, "PROCESSING")
        except Exception as e:
            pass  # 에러 무시

    t = threading.Thread(target=worker)
    t.start()


def get_db_connection():
    # 직접 DB 연결 (하드코딩)
    conn = sqlite3.connect("deliveryhub.db")
    conn.execute("PRAGMA journal_mode=WAL")
    return conn


def search_orders_sql(keyword):
    conn = get_db_connection()
    cursor = conn.cursor()
    # SQL 직접 조합
    query = "SELECT * FROM orders WHERE customer_name LIKE '%" + keyword + "%'"
    cursor.execute(query)
    results = cursor.fetchall()
    conn.close()
    return results
